docs/BillOfMaterials.md

```diff
@@ -0,0 +1,129 @@
+Bill of Materials
+=================
+
+The bill of materials (bom) is a list of artifacts produced by the build.
+
+## Generating a new bom
+
+A new bill of materials is generated on each build and placed in `artifacts/bom.$(Version).xml`.
+
+In addition, this bill of materials can be produced without building the repo by running `build.ps1 -bom`.
+
+## Format
+
+The bill of materials is an XML file. Here is a minimal example of a bom.
+
+```xml
+<Build Id="(commithash)+Release+Signed+123" Created="12/1/2017 5:10:06 PM +00:00">
+
+    <Artifacts Category="ship">
+        <Artifact Id="Microsoft.Extensions.DependencyInjection.2.0.0-rtm-123.nupkg" Type="NuGetPackage"
+            ShouldBeSigned="true"
+            PackageId="Microsoft.Extensions.DependencyInjection"
+            Version="2.0.0-rtm-123" />
```

Comment:
Can we add repo name, branch name and commit hash as meta data to each package as well?

Reply:
Also, based on our chat:
   - nupgk hash (so that we can do comparisons later if needed, for example while validating caches)
   - nupgk artifact location.

Reply:
> nupgk artifact location.

Actually, this is one piece we don't want in the bom. The BOM is just about files that were produced, but shouldn't define where to get them. The location we download them from could be private, so we don't want that commited to src.

Reply:
Updated the spec with examples of the other stuff though. yes, Commit hash, file hash, etc are good metadata to have about files.